### PR TITLE
feat: build no-deps

### PR DESCRIPTION
**Pull request by mattkrick**
# Description

Testing out GH Actions new larger runners.
This should build with `--no-deps`

## Demo

Look at the `build-docker` GH Action

### CODE REVIEW

**Comment by mattkrick** [comment]
@rafaelromcar-parabol 
Success using the larger runner. I scrapped some extra logic, we'll want to re-add that back in so it only runs this for tagged builds. Feel free to clone & hack away on this branch

**Comment by rafaelromcar-parabol** [comment]
**Context size and dockerignore** 

- I've added some stuff to the dockerignore because the context after runngin yarn build --no-deps was still very big and it was spending more than 80s on `#9 transferring context: 1.37GB 86.7s done`. You can see it in [this execution](https://github.com/ParabolInc/parabol/actions/runs/4984605257/jobs/8923156084). In sum up: now it is ignores all non required folders that start by a dot `.` and the folders `docs`, `node_modules`, `packages`, `scripts` and `static`. That way Docker doesn't even see those folders and do not need to charge them into its build context. This change took the workflow time from [8m17s](https://github.com/ParabolInc/parabol/actions/runs/4984605257/jobs/8923156084) to [5m17s](https://github.com/ParabolInc/parabol/actions/runs/4990577369/jobs/8935968901). Which is really huge! 5 minutes sounds like a good enough pipeline time and there is still room for improvement!

**Errors**
Right now, it isn't working. It's throwing:

```
ServerID 88
node:internal/modules/cjs/loader:1078
  throw err;
  ^

Error: Cannot find module '/home/node/parabol/scripts/migrate.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.16.0
```

We probably need to copy something else. You can check on the lines `101-103` on the `pipeline.dockerfile` what folders are copied to the Docker image.

I think we need some sync to test these real time.

=========
NOTES FOR THE FUTURE
- **Build consistency and security patches:** I changed the order in the pipeline.dockerfile because our builds aren't consistent right now. In our build process there is a part where we call dnf update -y and we upgrade packages. We could build our docker from the same commit twice in a week and get two different images. The second one would be upgraded to the last security patches, but this isn't a good practice. I need to think how to better manage this and make our builds consistent while keeping security at a good level. This would also reduce the build time by about 1m30s I estimate, but I have to test it to validate my theory. This is for another PR, but just wanted to share my thoughts here.

**Comment by rafaelromcar-parabol** [comment]
Turns out I just didn't use the correct command to run the database migrations. I was using `scripts/migrate.js`... I'm now using `dist/migrate.js` and the error message is the following:

```
ServerID 103
node:internal/validators:163
    throw new ERR_INVALID_ARG_TYPE(name, 'string', value);
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at new NodeError (node:internal/errors:399:5)
    at validateString (node:internal/validators:163:11)
    at Object.join (node:path:1172:7)
    at 888682 (/home/node/parabol/dist/migrate.js:1:5083202)
    at r (/home/node/parabol/dist/migrate.js:1:5371630)
    at /home/node/parabol/dist/migrate.js:1:5372071
    at Object.<anonymous> (/home/node/parabol/dist/migrate.js:1:5372100)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32) {
  code: 'ERR_INVALID_ARG_TYPE'
}

Node.js v18.16.0
```

File: docker/parabol-ubi/docker-build/dockerfiles/pipeline.dockerfile [hunk 1]
```diff
@@ -98,7 +97,9 @@ RUN  if [ "$_SECURITY_ENABLED" = "true" ]; then \
 
 RUN rm -rf /security/
 
-COPY --chown=node . ${HOME}/parabol
+COPY --chown=node .env.example ${HOME}/parabol/.env.example
+COPY --chown=node build ${HOME}/parabol/build
+COPY --chown=node dist ${HOME}/parabol/dist
```
[the remark this answers is not in the document]
Comment (Dschoordsch) [reply docker/parabol-ubi/docker-build/dockerfiles/pipeline.dockerfile:107]: Copy that one as well :D

[the remark this answers is not in the document]
Comment (rafaelromcar-parabol) [reply docker/parabol-ubi/docker-build/dockerfiles/pipeline.dockerfile:107]: After adding the queryMap.json to the parabol folder, it looks like everything worked just fine!!!

```
Added 0 records to the queryMap
```
And the application is up and running on that last container 🥇 Thanks Georg!

[the remark this answers is not in the document]
Comment (mattkrick) [reply docker/parabol-ubi/docker-build/dockerfiles/pipeline.dockerfile:107]: Will fix this in https://github.com/ParabolInc/parabol/issues/8202

**Comment by mattkrick** [comment]
After testing in #8204 we confirmed that it's safe to remove `.dockerignore` and it does not increase build time

**Pull request merged**